src/gt_frontend/frontend.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+#
+# Eve Toolchain - GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2020, CSCS - Swiss National Supercomputing Center, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+import ast
+import inspect
+import textwrap
+
+import devtools
+from gt_frontend.gtscript_to_gtir import (
+    GTScriptToGTIR,
+    NodeCanonicalizer,
+    SymbolResolutionValidation,
+    SymbolTable,
+    TemporaryFieldDeclExtractor,
+    VarDeclExtractor,
+)
+from gt_frontend.py_to_gtscript import PyToGTScript
+
+from gtc import common
+from gtc.unstructured.gtir_to_nir import GtirToNir
+from gtc.unstructured.nir_passes.merge_horizontal_loops import find_and_merge_horizontal_loops
+from gtc.unstructured.nir_to_usid import NirToUsid
+from gtc.unstructured.usid_codegen import UsidGpuCodeGenerator
+
+
+class GTScriptCompilationTask:
```

Comment:
The intermediate results are stored since they are used in the unit tests. I spent absolutely no time in making this clean since the complete class will disappear with the `PassManager` to be introduced in Eve and the build stages approach @DropD introduced in GT4Py. I've added a todo comment in the meantime.